### DISCUSSION TITLE
How do I remove default `<leader>wm` and map `<leader>w` to something custom without it being labeled as `+windows`?

### DISCUSSION THREAD

**Discussion by jpridemore-allegion** [LazyVim/LazyVim#4062] [Q&A] [answered]
I'm trying to make it so that `<leader>w` is mapped, in normal mode, to `<cmd>update<cr>`. Because LazyVim wants to have `+windows` hotkeys there, this has proven _very_ difficult.  All of the stuff on `<leader>w` is already effectively on `<C-w>` and then a key, and I'm used to those from before LazyVim, so I want to remove the `<leader>w` ones and have it go back to writing the file like I'm comfortable with. Right now, in my `lua/config/keymaps.lua` file, I have this:

```lua
local unmap = vim.keymap.del

-- ...

-- # Windows
LazyVim.toggle.map("<C-w>m", LazyVim.toggle.maximize)
unmap("n", "<leader>ww")
unmap("n", "<leader>wd")
unmap("n", "<leader>w-")
unmap("n", "<leader>w|")
unmap("n", "<leader>-")
unmap("n", "<leader>|")
-- TODO: How to make this take?!
unmap("n", "<leader>wm")
```

And in my `lua/plugins/which-key.lua` I have this:

```lua
return {
  "folke/which-key.nvim",
  opts = {
    spec = {
      plugins = { spelling = true },
      {
        mode = { "n", "v" },
        { "g", group = "goto" },
        { "gs", group = "surround" },
        { "z", group = "fold" },
        { "[", group = "prev" },
        { "]", group = "next" },
        { "<leader><tab>", group = "tabs" },
        { "<leader>b", group = "buffer" },
        { "<leader>c", group = "code" },
        { "<leader>f", group = "file/find" },
        { "<leader>g", group = "git" },
        { "<leader>gh", group = "hunks" },
        { "<leader>q", group = "quit/session" },
        { "<leader>s", group = "search" },
        { "<leader>u", group = "ui", icon = { icon = "󰙵 ", color = "cyan" } },
        { "<leader>x", group = "diagnostics/quickfix", icon = { icon = "󱖫 ", color = "green" } },
        { "<leader>v", group = "tasks" },
        { "<leader>w", desc = "Write" },
        -- TODO: Do I need this?
        -- { "<leader>w", "<cmd>update<cr>", desc = "write", group = nil },
        { "<leader>y", group = "copy" },
      },
    },
  },
}
```

Which is effectively a copy and paste of the which key for LazyVim but trying to get rid of that `+windows` group.

If I hit my `<leader>` key, then I see the group `write` correctly but if I then hit `w`, I'll see the `Enable Maximize` toggle instead of it saving the file. How do I undo what `LazyVim.toggle.map("<leader>wm", LazyVim.toggle.maximize)` has done in a way that'll actually take? I don't see a `LazyVim.toggle.unmap` equivalent, and I _do_ think I'm fully updated (including matching the `which-key` v3 config shape).

Full Configuration: https://github.com/Riobe/config-nvim/tree/main

**Comment by folke** [comment, marked as answer]
You'll have to remove the mapping from which-key as well.

`{ "<leader>wm", hidden = true }`

**Reply by jpridemore-allegion** [reply]
Thank you! 

```lua
        { "<leader>w", desc = "write" },
        { "<leader>wm", hidden = true },
```
In my `lua/plugins/which-key.lua`

and

```lua
LazyVim.toggle.map("<C-w>m", LazyVim.toggle.maximize)
unmap("n", "<leader>wm")
```
In my `lua/config/keymaps.lua` was exactly what I needed to be able to move it to `<C-w>m` and set `<leader>w` to what I wanted. I appreciate it!

**Reply by utkarshgupta137** [reply]
This isn't working for me... `<leader>w` works if I press is fast enough (I think it bypasses which-key), but if I'm not fast enough, I just get a menu with a single option "Enable Maximize".
<img width="2056" alt="image" src="https://github.com/user-attachments/assets/19e1f8b5-05e8-4ea8-ae90-f23458a95c4c">

**Reply by leikoilja** [reply]
@utkarshgupta137, have you found a solution? I have been struggling with the same issue when trying to bind `<leader>ww` to save a file. 
if i click it quick enough - it saves the file, but if i not fast enough it changes windows (as a default `<C-w>w` would do...
i tried everything: unmapping, reassigning... none has helped

**Reply by utkarshgupta137** [reply]
> @utkarshgupta137, have you found a solution? I have been struggling with the same issue when trying to bind `<leader>ww` to save a file. 
> if i click it quick enough - it saves the file, but if i not fast enough it changes windows (as a default `<C-w>w` would do...
> i tried everything: unmapping, reassigning... none has helped

Nope, still struggling.

**Reply by leikoilja** [reply]
@utkarshgupta137 , phew, just managed to make it work, [here](https://github.com/LazyVim/LazyVim/discussions/1427#discussioncomment-11460296) is my solution

**Reply by utkarshgupta137** [reply]
Your solution only worked if you wanted `<leader>ww`. I want it to work with `<leader>w` directly. I found a [fix](https://github.com/LazyVim/LazyVim/discussions/1427#discussioncomment-11462021) today.

**Comment by GreenSleeper** [comment]
Is there any simple way to unmap all keymaps prefixed with `<leader>w`? now there are too many keymaps under it.

**Reply by dpetka2001** [reply]
```lua
{
  "folke/which-key.nvim",
  opts = {
    spec = {
      { "<leader>w", proxy = false },
    },
  },
},
```
should do it I believe.

**Reply by GreenSleeper** [reply]
this works, thank you.
